addons/compute_worker/gpu_uniforms/gpu_struct.gd

```diff
@@ -71,13 +72,20 @@ func set_uniform_data(rd: RenderingDevice, data: Array) -> void:
 ## Encode the contents of the passed in `data` Array to PackedByteArray. 
 ## Contents of `data` must match the order and data types defined in `struct_data`.
 func encode_struct(data: Array, init: bool = false) -> PackedByteArray:
-	
 	var arr: PackedByteArray = PackedByteArray()
 	var data_index = 0
 	
 	for type_obj in struct_data:
 		
 		match typeof(type_obj):
+			TYPE_PACKED_VECTOR3_ARRAY:
+				arr.append_array(vec3_array_to_byte_array(data[data_index]))
+			TYPE_PACKED_VECTOR2_ARRAY:
+				arr.append_array(vec2_array_to_byte_array(data[data_index]))
+			TYPE_VECTOR2I:
+				arr.append_array(vec2i_to_byte_array(data[data_index]))
```

Comment:
`TYPE_VECTOR2I` and `TYPE_VECTOR2` require different alignments when inside a struct. Like with arrays, alignments for struct members are rounded up to the alignment of a `vec4`. So for these two types, we'll need separate functions to get the byte arrays with the extra padding (+2 components).